src/input_common/gcadapter/gc_poller.cpp

```diff
@@ -15,36 +15,43 @@ namespace InputCommon {
 
 class GCButton final : public Input::ButtonDevice {
 public:
-    explicit GCButton(u32 port_, int button_, const GCAdapter::Adapter* adapter)
+    explicit GCButton(u32 port_, int button_, GCAdapter::Adapter* adapter)
         : port(port_), button(button_), gcadapter(adapter) {}
 
     ~GCButton() override;
 
     bool GetStatus() const override {
         if (gcadapter->DeviceConnected(port)) {
-            return gcadapter->GetPadState()[port].buttons.at(button);
+            return gcadapter->GetPadState(port).buttons & button;
```

Comment:
```suggestion
            return gcadapter->GetPadState(port).buttons & button != 0;
```
Makes it a bit more expressive on the intent